src/lager_loggly_backend.erl

```diff
@@ -68,22 +69,24 @@ handle_call(_Request, State) ->
     {ok, ok, State}.
 
 %% @private
-handle_event({log, Level, {_Date, _Time}, [_LvlStr, Loc, Message]}, State)
-        when Level =< State#state.level ->
-    Payload = jsx:to_json([
-                            {<<"identity">>, any_to_binary(State#state.identity)}
-                            ,{<<"level">>, convert_level(Level)}
-                            ,{<<"location">>, any_to_binary(Loc)}
-                            ,{<<"message">>, any_to_binary(Message)}
-                          ]),
-    Request = {State#state.loggly_url, [], "application/json", Payload},
-    RetryTimes = State#state.retry_times,
-    RetryInterval = State#state.retry_interval,
-
-    %% Spawn a background process to handle sending the payload.
-    %% It will recurse until the payload has ben successfully sent.
-    spawn(?MODULE, deferred_log, [Request, RetryTimes, RetryInterval]),
-    {ok, State};
+handle_event({log, Message}, #state{level=Level} = State) ->
+    case lager_util:is_loggable(Message, Level, ?MODULE) of
+        true ->
+            Payload = jsx:encode(cons_metadata_to_binary_proplist(lager_msg:metadata(Message), [
```

Comment:
The function `fun cons_metadata_to_binary_proplist/2` will convert every dynamic meta data (`pid`, `file`, `line`, `module`, ...) appropriate from the message and include them to the final payload.